src/SOS/SOS.Hosting/SOSHost.cs

```diff
@@ -455,6 +462,93 @@ public void ExecuteCommand(string command, string arguments)
             return E_FAIL;
         }
 
+        internal unsafe int ReadVirtualForWindows(
+            IntPtr self,
+            ulong address,
+            IntPtr buffer,
+            uint bytesRequested,
+            uint* pbytesRead)
+        {
+            if (DataReader.ReadMemory(address, buffer, unchecked((int)bytesRequested), out int bytesRead))
```

Comment:
In my ideal world CLRMD would also have access to the image mapped memory so that managed and native authored commands wouldn't have any discrepancies in their view of memory. If that is challenging doing SOS only seems like a fine first step.

Reply:
I think this is a good idea, but too much to do right now.  I created this issue to track: https://github.com/dotnet/diagnostics/issues/436